lib/ModelMixins/CatalogMemberMixin.ts

```diff
@@ -180,16 +181,32 @@ function CatalogMemberMixin<T extends Constructor<CatalogMember>>(Base: T) {
             );
             const value = dim.options.find(o => o.id === selectedId)?.value;
             if (isDefined(value)) {
-              updateModelFromJson(this, stratumId, value).raiseError(
+              const result = updateModelFromJson(this, stratumId, value);
+              result.raiseError(
                 this.terria,
                 `Failed to update catalog item ${getName(this)}`
               );
+
+              // If no error and modelDimensionCallLoadMapItems is true, then call loadMapItems
+              if (
+                !result.error &&
+                this.modelDimensionCallLoadMapItems &&
+                MappableMixin.isMixedInto(this)
+              ) {
+                this.loadMapItems().then(loadMapItemsResult => {
+                  loadMapItemsResult.raiseError(this.terria);
+                });
+              }
```

Comment:
Doubt: shouldn't this be called always? Considering in most cases the reload will happen only if any traits that affect loading has changed :thinking:

Reply:
Yeah good point